### ISSUE TITLE
Magento 1.9 not calculating Tax Amount

### ISSUE THREAD

**Issue by jefflei** [wyrdough/taxcloud-magento#2]
In Magento 1.9 it seems like the tax rate is correctly identified, but the tax calculation is not calculating correctly.  I am getting $0 for tax.

Please can someone help.

**Comment by wyrdough** [comment]
Does the order show up on TaxCloud at all?

When the calculation for a given order is made it should show up as an incomplete order on the TC site. If the order data isn't making it there, it may show a single dummy item which is used to calculate the rate for display, but without the full order details it does not get the correct amount to charge back from the web service.

Also, please provide any relevant PHP or Magento logging  that may have been generated.

I don't have a store running 1.9, so I'll need your help to figure out where the problem lies.

**Comment by jefflei** [comment]
The API is working perfectly on TaxCloud, all fields are populated.

When I look at the orders, see attached image.  I see the tax percent is populated, but it is not updating the tax amount.

I am novice at this, so am not sure how to get any PHP or Magento logging.  If you can point me in the right direction, I would be glad to help.

The store site is:  store.creatisticinc.com..

![orders _ sales _ magento admin](https://cloud.githubusercontent.com/assets/15663212/10978321/22e733c0-83ab-11e5-9db4-f87780cf641f.png)

**Comment by wyrdough** [comment]
Just to confirm, the shipping address is a US address and the TaxCloud interface shows tax due for this order, correct? If so, I suspect there are changes between 1.6 and 1.9 in how the calculated tax amount for an order is stored and I'll need to look at the code for the new version, which will take a bit.

**Comment by jefflei** [comment]
Yes you are correct that the tax rate is calculating correctly for the US address where our site is registered.  On TaxCloud, it actually shows the correct tax calculation and amount.
![image](https://cloud.githubusercontent.com/assets/15663212/10987691/a2eba36e-83ea-11e5-81ed-aba60839be30.png)

What I need help with is how to transfer the tax rate \* selling price to the tax amount field.  Somehow this is not working correctly.

Avalara have a working extension which you might be able to look at as well.

**Comment by jefflei** [comment]
I spoke with Taxcloud and they referred us to their developer portal (dev.taxcloud.com).  They also indicated that they pass back tax amounts, not tax rates, which might help on the fix.

**Comment by Premmy** [comment]
I believe I am using a different package (https://github.com/taxcloud/Magento) and wondering if this is a better option?  I have been able to setup the configuration in Magento 1.9.3 but it does not seem to be doing any API calls so if anyone has this working I will definitely try this code base.
